dot/network/notifications.go

```diff
@@ -20,7 +20,9 @@ const handshakeTimeout = time.Second * 10
 
 // Handshake is the interface all handshakes for notifications protocols must implement
 type Handshake interface {
-	NotificationsMessage
+	Message
+	Type() byte
+	IsValidHandshake() bool
```

Comment:
do we even need an interface for this anymore? Could we not do type checks on the types that are handshake messages? Or even use the constraint feature in go generics.

Reply:
`IsValidHandshake` validates the handshake as well. It does more that type checking.